saleor/warehouse/management.py

```diff
@@ -382,7 +401,15 @@ def get_order_lines_with_track_inventory(
 @traced_atomic_transaction()
 def deallocate_stock_for_order(order: "Order"):
     """Remove all allocations for given order."""
-    allocations = Allocation.objects.filter(
-        order_line__order=order, quantity_allocated__gt=0
-    ).select_for_update(of=("self",))
+    allocations = (
+        Allocation.objects.filter(order_line__order=order, quantity_allocated__gt=0)
```

Comment:
```suggestion
        Allocation.objects.filter(order_line_order_id=order.id, quantity_allocated__gt=0)
```

Reply:
Couldn't do it like that since order_line x order is still a relation. I got rid of the join though.